example/spec/kinesisTests/KinesisTestTriggerSpec.js

```diff
@@ -0,0 +1,154 @@
+
+const { loadConfig } = require('../helpers/testUtils');
+const {
+  createOrUseTestStream,
+  putRecordOnStream,
+  waitForActiveStream,
+  waitForTestSfStarted
+} = require('../helpers/kinesisHelpers');
+
+const record = require('../../data/records/L2_HR_PIXC_product_0001-of-4154.json');
+
+const granuleId = record.product.name;
+const recordIdentifier = randomString();
+record.identifier = recordIdentifier;
+
+const testConfig = loadConfig();
+
+const lambdaStep = new LambdaStep();
+const streamName = testConfig.streamName;
+
+
+const recordFile = record.product.files[0];
+const expectedTranslatePayload = {
+  granules: [
+    {
+      granuleId: record.product.name,
+      files: [
+        {
+          path: 'cumulus-test-data/pdrs',
+          url_path: recordFile.uri,
+          bucket: record.bucket,
+          name: recordFile.name,
+          size: recordFile.size
+        }
+      ]
+    }
+  ]
+};
+
+const fileData = expectedTranslatePayload.granules[0].files[0];
+const filePrefix = `file-staging/${testConfig.stackName}/L2_HR_PIXC`;
+
+const fileDataWithFilename = {
+  ...fileData,
+  filename: `s3://${testConfig.buckets.private.name}/${filePrefix}/${recordFile.name}`,
+  bucket: testConfig.buckets.private.name,
+  url_path: '',
+  fileStagingDir: filePrefix
+};
+
+const expectedSyncGranulesPayload = {
+  granules: [
+    {
+      granuleId: granuleId,
+      files: [fileDataWithFilename]
+    }
+  ]
+};
+
+
+// When Cumulus is configured to trigger a CNM workflow from a Kinesis stream and a message appears on the stream, Cumulus triggers the workflow
+describe('The Cloud Notification Mechanism Kinesis workflow', () => {
+  const maxWaitTime = 1000 * 60 * 2;
+  let workflowExecution;
+  let executionStatus;
+  let s3FileHead;
+
+  afterAll(async () => {
+    await s3().deleteObject({
+      Bucket: testConfig.buckets.private.name,
+      Key: `${filePrefix}/${fileData.name}`
+    }).promise();
+  });
+
+  beforeAll(() => {
+    this.workflowExecution = null;
+  });
+
+  it('finds or creates a test stream.', async () => {
+    await createOrUseTestStream(streamName);
+  });
+
+  it('waits for the stream to be active.', async () => {
+    await waitForActiveStream(streamName);
+  });
+
+  it('places a record on the stream.', async () => {
+    await putRecordOnStream(streamName, record);
+  });
+
+  it('waits for the triggered step function to start', async () => {
+    workflowExecution = await waitForTestSfStarted(recordIdentifier, maxWaitTime);
+    this.workflowExecution = workflowExecution;
+  });
+
+  it('finds a valid workflow execution.', () => {
+    expect(this.workflowExecution).not.toBe(undefined);
+  });
+
+  it('waits for step function to complete', async () => {
+    executionStatus = await waitForCompletedExecution(this.workflowExecution.executionArn);
+  });
+
+  it('executes successfully', () => {
+    expect(executionStatus).toEqual('SUCCEEDED');
+  });
```

Comment:
I think most of tests `The Cloud Notification Mechanism Kinesis workflow` should be in a beforeAll block - most are not testing anything. Only `finds a valid workflow execution.` and `executes successfully` are tests.

Reply:
Same with some other tests in this PR, e.g. `Closes a kinesis test stream`

Reply:
So is that actually bad?  it was a style thing for me.  Originally all of the setup was in the beforeAll, and when something failed it just thew it's exception and then started running the tests and I had a really hard time figuring out where the error was coming from.

I like that that each step in the setup was it's own step that could succeed or fail and that the steps tell you what's happening in real time as your watching them.    But if this is bad form, I can move them back into a before all.  (But, I really didn't like the way after beforeAll failed, the tests just tried to run anyway and just fail with errors unrelated to the problem that occurred.)

Reply:
I know what you mean - my preferred adaptation would be to make sure the test suite exits if the beforeAll block doesn't complete successfully. I looked into this a bit and it looks like jasmine doesn't offer a `fail fast` option (ref: https://stackoverflow.com/questions/11562817/does-jasmine-node-offer-any-type-of-fail-fast-option, which links to a bunch of related jasmine github issues). I would be interested to know if the `process.exit(1)` solution works.

Alternatively, if we keep these as unit tests, we should actually assert something - what do you think?

Reply:
Ok, if we leave as is, you'd like to see `expects().something()` in each  `it` "test".  Or you'd like to see all of the non-test setup in a beforeAll()?  
I bet you have a preference for the latter?  and have to go look to see if I prefer the first.